### DISCUSSION TITLE
Angular projects generated by coveo cli uses multiple deprecated features

### DISCUSSION THREAD

**Discussion by akaustav** [coveo/cli#282] [Q&A] [answered]
Hi,
I was able to generate an Angular 11 project using the [`coveo ui:create:angular NAME`](https://github.com/coveo/cli/tree/master/packages/cli#coveo-uicreateangular-name) command. The project connects to a test / development organization which I created by following instructions from a [previous discussion](https://github.com/coveo/cli/discussions/245). While the project works in my local machine, I noticed that the code imports a few classes, constants, namespaces, etc. from the `@coveo/headless` library which have now been marked as deprecated. Is there any way to get an example Angular 11 application which does NOT rely on deprecated features from the `@coveo/headless` library?

### Examples of deprecated imports

- https://github.com/coveo/cli/blob/47c98ccd9ee632aa35975ec2781efe2b1d6d3a0e/packages/angular/src/ng-add-setup-project/files/src/app/engine.service.ts#L2

- https://github.com/coveo/cli/blob/47c98ccd9ee632aa35975ec2781efe2b1d6d3a0e/packages/angular/src/ng-add-setup-project/files/src/app/search-page/search-page.component.ts#L2

### Deprecation notes from `@coveo/headless` code in the [`coveo/ui-kit`](https://github.com/coveo/ui-kit) repo

- `HeadlessEngine` class: https://github.com/coveo/ui-kit/blob/9e00f131bfe2f3437345f09e88a42896cd719998/packages/headless/src/app/headless-engine.ts#L77-L86

- `searchAppReducers` constant: https://github.com/coveo/ui-kit/blob/9e00f131bfe2f3437345f09e88a42896cd719998/packages/headless/src/app/search-app-reducers.ts#L36-L41

- `AnalyticsActions` namespace: https://github.com/coveo/ui-kit/blob/9e00f131bfe2f3437345f09e88a42896cd719998/packages/headless/src/features/analytics/index.ts#L11-L14

- `SearchActions` namespace: https://github.com/coveo/ui-kit/blob/9e00f131bfe2f3437345f09e88a42896cd719998/packages/headless/src/features/index.ts#L305-L308

Ameet

**Comment by olamothe** [comment, marked as answer]
Hello, 

Coveo Headless is currently on a transition path to the next major version, which is why you are seeing these deprecation message.

The current setup, with the version of @coveo/headless generated for your project will continue to work for the current major version.

We are currently doing a release (v.1.5.0) of the CLI which migrates and support the next major version of Headless. This should be available either by EOD or tomorrow.

You can upgrade the CLI using `coveo update` command.

Thank you for reporting !

**Reply by akaustav** [reply]
Thanks for the update, @olamothe. I tried upgrading using `coveo update` command now - didn't work. Posted the information in a separate discussion https://github.com/coveo/cli/discussions/285.
